cpp/cmake/thirdparty/get_spdlog.cmake

```diff
@@ -19,15 +19,6 @@ function(find_and_configure_spdlog)
     rapids_cpm_spdlog(FMT_OPTION "EXTERNAL_FMT_HO" INSTALL_EXPORT_SET rmm-exports)
     rapids_export_package(BUILD spdlog rmm-exports)
 
-    if(spdlog_ADDED)
-        rapids_export(
-                BUILD spdlog
-                EXPORT_SET spdlog
-                GLOBAL_TARGETS spdlog spdlog_header_only
-                NAMESPACE spdlog::)
-        include("${rapids-cmake-dir}/export/find_package_root.cmake")
-        rapids_export_find_package_root(BUILD spdlog [=[${CMAKE_CURRENT_LIST_DIR}]=] EXPORT_SET rmm-exports)
-    endif()
```

Comment:
context for this change: https://github.com/rapidsai/rapids-cmake/issues/387#issuecomment-2077550701